src/wp-admin/includes/image.php

```diff
@@ -341,6 +341,18 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 				 */
 				if ( $scale_down ) {
 					$saved = $editor->save( $editor->generate_filename( 'scaled' ) );
+				} elseif ( $convert ) {
+					/*
+					 * Generate a new file name for the converted image.
+					 *
+					 * As the image file name will be unique due to the changed file extension,
+					 * it does not need a suffix to be unique. However, the generate_filename method
+					 * does not allow for an empty suffix, so the "-converted" suffix is required to
+					 * be added and subsequently removed.
+					 */
+					$converted_file_name = $editor->generate_filename( 'converted' );
+					$converted_file_name = preg_replace( '/-converted\.[a-z0-9]+$/i', '.', $converted_file_name );
```

Comment:
This regex confuses me. Why is the `\.[a-z0-9]+$` portion in there? Wouldn't that also replace (effectively remove) the file extension? Since the tests pass, probably this is me not understanding how this regex works, but would be great if you could clarify.

Reply:
@felixarntz Yes, that's a mistake and it looks like the save method came to my rescue. I've pushed a fix in 9f84fcd76b9400838fb67bb77af84b88982086a2